ecosystem/indexer-grpc/indexer-grpc-utils/src/in_memory_cache.rs

```diff
@@ -254,23 +258,34 @@ fn spawn_update_task<C>(
             }
             let end_version = std::cmp::min(
                 current_latest_version,
-                in_cache_latest_version + 10 * MAX_REDIS_FETCH_BATCH_SIZE as u64,
+                in_memory_cache_latest_version + 10 * MAX_REDIS_FETCH_BATCH_SIZE as u64,
             );
-            let versions_to_fetch = (in_cache_latest_version..end_version).collect();
+            let versions_to_fetch = (in_memory_cache_latest_version..end_version).collect();
             let transactions = batch_get_transactions(&mut conn, versions_to_fetch, storage_format)
                 .await
                 .unwrap();
             // Ensure that transactions are ordered by version.
             let mut newly_added_bytes = 0;
             for (ind, transaction) in transactions.iter().enumerate() {
-                if transaction.version != in_cache_latest_version + ind as u64 {
+                if transaction.version != in_memory_cache_latest_version + ind as u64 {
                     panic!("Transactions are not ordered by version");
                 }
                 newly_added_bytes += transaction.encoded_len() as u64;
             }
             for transaction in transactions {
                 cache.insert(transaction.version, Arc::new(transaction));
             }
+            tracing::debug!(
+                in_memory_cache_first_version = in_memory_cache_first_version,
```

Comment:
wait the `in_memory_cache_first_version` could have changed because we just inserted some items into it ealier.